dart/lib/src/transport/task_queue.dart

```diff
@@ -0,0 +1,27 @@
+import 'dart:async';
+
+import '../../sentry.dart';
+
+typedef Task<T> = Future<T> Function();
+
+class TaskQueue<T> {
+  TaskQueue(this._maxQueueSize, this._logger);
+
+  final int _maxQueueSize;
+  final SentryLogger _logger;
+
+  int _queueCount = 0;
+
+  Future<T> enqueue(Task<T> task, T fallbackResult) async {
+    if (_queueCount >= _maxQueueSize) {
+      _logger(SentryLevel.warning,
+          'Task dropped due to backpressure. Avoid capturing in a tight loop.');
+      return fallbackResult;
+    } else {
+      _queueCount++;
+      final result = await task();
```

Comment:
If a task throws an exception, it seems that `_queueCount` would not be decremented

maybe this is safer:
```
  _queueCount++;
  try {
    final result = await task();
    return result;
  } finally {
    _queueCount--;
  }
```